src/nvim/ops.c

```diff
@@ -2791,7 +2792,8 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
     xfree(reg->y_array);
   }
 
-  if (message && (p_ch > 0 || ui_has(kUIMessages))) {  // Display message about yank?
+  // TODO(Shougo): Use ext_messages instead
+  if (message && ui_has_messages()) {  // Display message about yank?
```

Comment:
This is another `smsg()` case. Should not need to special-case this here. Should be handled in messages.c, and display of the message is controlled by the user via 'shortmess' etc.